ethstorage/p2p/protocol/syncclient.go

```diff
@@ -596,15 +596,59 @@ func (s *SyncClient) RequestL2Range(start, end uint64) (uint64, error) {
 		if err != nil {
 			return 0, err
 		}
-		_, _, _, err = s.onResult(packet.Blobs)
+		_, _, inserted, err := s.onResult(packet.Blobs)
 		if err != nil {
 			return 0, err
 		}
-		return id, nil
+
+		return uint64(len(inserted)), nil
 	}
 	return 0, fmt.Errorf("no peer can be used to send requests")
 }
 
+func (s *SyncClient) FetchBlob(kvIndex uint64, commit common.Hash) ([]byte, error) {
+	if len(s.peers) == 0 {
+		return []byte{}, fmt.Errorf("no peer can be used to send requests")
+	}
+	for _, pr := range s.peers {
+		var packet BlobsByListPacket
+		var payload *BlobPayload = nil
+
+		_, err := pr.RequestBlobsByList(rand.Uint64(), s.storageManager.ContractAddress(), kvIndex/s.storageManager.KvEntries(), []uint64{kvIndex}, &packet)
+		if err != nil {
+			log.Warn("FetchBlob failed", "error", err)
+		}
+
+		for _, val := range packet.Blobs {
+			if val.BlobIndex != kvIndex {
+				continue
+			}
+			if val.BlobCommit.Cmp(commit) != 0 {
```

Comment:
We should compare the part of `commit[0:HashSizeInContract]` instead of all hash